pkg/api/util/ipc_endpoint_test.go

```diff
@@ -0,0 +1,212 @@
+}
+
+func TestNewIPCEndpointWithCloseConnection(t *testing.T) {
+	conf := pkgconfigmodel.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
+
+	// create a fake auth token
+	authTokenFile, err := os.CreateTemp("", "")
+	assert.NoError(t, err)
+	authTokenPath := authTokenFile.Name()
+	os.WriteFile(authTokenPath, []byte("0123456789abcdef0123456789abcdef"), 0640)
+
+	// set minimal configuration that IPCEndpoint needs
+	conf.Set("auth_token_file_path", authTokenPath, pkgconfigmodel.SourceAgentRuntime)
+	conf.Set("cmd_host", "localhost", pkgconfigmodel.SourceAgentRuntime)
+	conf.Set("cmd_port", "6789", pkgconfigmodel.SourceAgentRuntime)
+
+	// test constructing with the CloseConnection option
+	end, err := NewIPCEndpoint(conf, "test/api", WithCloseConnection(true))
+	assert.NoError(t, err)
+	assert.True(t, end.closeConn)
+}
+
+func TestIPCEndpointGet(t *testing.T) {
+	gotURL := ""
+	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		gotURL = r.URL.String()
+		_, _ = io.ReadAll(r.Body)
+		w.Write([]byte("ok"))
+	}))
+	defer ts.Close()
+
+	conf := createConfig(t, ts)
+	end, err := NewIPCEndpoint(conf, "test/api")
+	assert.NoError(t, err)
+
+	// test that DoGet will hit the endpoint url
+	res, err := end.DoGet()
+	assert.NoError(t, err)
+	assert.Equal(t, res, []byte("ok"))
+	assert.Equal(t, gotURL, "/test/api")
+}
+
+func TestIPCEndpointGetWithValues(t *testing.T) {
+	gotURL := ""
+	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		gotURL = r.URL.String()
+		_, _ = io.ReadAll(r.Body)
+		w.Write([]byte("ok"))
+	}))
+	defer ts.Close()
+
+	conf := createConfig(t, ts)
+	// set url values for GET request
+	v := url.Values{}
+	v.Set("verbose", "true")
+
+	// test construction with option for url.Values
+	end, err := NewIPCEndpoint(conf, "test/api")
+	assert.NoError(t, err)
+
+	// test that DoGet will use query parameters from the url.Values
+	res, err := end.DoGet(WithValues(v))
+	assert.NoError(t, err)
+	assert.Equal(t, res, []byte("ok"))
+	assert.Equal(t, gotURL, "/test/api?verbose=true")
+}
+
+func TestIPCEndpointGetWithHostAndPort(t *testing.T) {
+	gotURL := ""
+	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		gotURL = r.URL.String()
+		_, _ = io.ReadAll(r.Body)
+		w.Write([]byte("ok"))
+	}))
+	defer ts.Close()
+
+	conf := createConfig(t, ts)
+	// modify the config so that it uses a different setting for the cmd_host
+	conf.Set("process_config.cmd_host", "localhost", pkgconfigmodel.SourceAgentRuntime)
+
+	// test construction with alternate values for the host and port
+	end, err := NewIPCEndpoint(conf, "test/api", WithHostAndPort(conf.GetString("process_config.cmd_host"), conf.GetInt("cmd_port")))
+	assert.NoError(t, err)
+
+	res, err := end.DoGet()
+	assert.NoError(t, err)
+	assert.Equal(t, res, []byte("ok"))
+	assert.Equal(t, gotURL, "/test/api")
```

Comment:
Are we testing that the host and port are different that the default values?

Reply:
Hmmm, you're right that this test is not very clear on what it was testing. Previously it was just exercising that `WithHostAndPort` "works" when passed to `NewIPCEndpoint`. I have changed it so that it validating the URL's host instead.